In [46]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
df = pd.read_csv('ForeignGifts_edu.csv')

In [47]:
df.value_counts('Gift Type')

Gift Type
Contract         17274
Monetary Gift    10936
Real Estate         11
Name: count, dtype: int64

The three different classes of gift type are contract, monetary gift, and real estate. A contract is a binding agreement between the donor and the university with specific instructions from the donor as to how the gift can be used. A monetary gift is a donation which can be used by the university for any purpose. A real estate gift is a land donation with the intention of having the university build on the land.

In [48]:
df_sorted_country = df.groupby(df['Country of Giftor'])
total_by_country = df_sorted_country['Foreign Gift Amount'].sum()
total_by_country.sort_values(ascending = False)

Country of Giftor
QATAR           2706240869
ENGLAND         1464906771
CHINA           1237952112
SAUDI ARABIA    1065205930
BERMUDA          899593972
                   ...    
IRAN                    80
ST. LUCIA               50
CROATIA                 26
SLOVAKIA                20
ERITREA                  3
Name: Foreign Gift Amount, Length: 155, dtype: int64

In [49]:
total_donations_country = df['Country of Giftor'].value_counts()
print(total_donations_country)

Country of Giftor
ENGLAND                  3655
CHINA                    2461
CANADA                   2344
JAPAN                    1896
SWITZERLAND              1676
                         ... 
ROMANIA                     1
TUNISIA                     1
ST. KITTS-NEVIS             1
MARSHAL ISLANDS (THE)       1
ANTIGUA                     1
Name: count, Length: 155, dtype: int64


In [50]:
country_avg = total_by_country / total_donations_country
country_avg.sort_values(ascending=False)

Country of Giftor
BERMUDA      7.688837e+06
ECUADOR      3.985140e+06
QATAR        3.905109e+06
ANTIGUA      3.435545e+06
JERSEY       3.064589e+06
                 ...     
ST. LUCIA    5.000000e+01
IRAN         4.000000e+01
CROATIA      2.600000e+01
SLOVAKIA     2.000000e+01
ERITREA      3.000000e+00
Length: 155, dtype: float64

In [51]:
df_sorted_uni = df.groupby(df['Institution Name'])
total_by_uni = df_sorted_uni['Foreign Gift Amount'].sum()
total_by_uni.sort_values(ascending = False)

Institution Name
Carnegie Mellon University               1477922504
Cornell University                       1289937761
Harvard University                        954803610
Massachusetts Institute of Technology     859071692
Yale University                           613441311
                                            ...    
Jacksonville State University                250000
Metropolitan State University                235439
University of California, Merced             182514
University of Central Oklahoma               160288
University of Jamestown                         500
Name: Foreign Gift Amount, Length: 318, dtype: int64

In [52]:
total_donations_uni = df['Institution Name'].value_counts()
print(total_donations_uni)

Institution Name
University of California, Los Angeles          3916
California Institute of Technology             2502
Columbia University in the City of New York    1127
Ohio State University (The)                    1014
Johns Hopkins University                        899
                                               ... 
University of Wisconsin - Platteville             1
Des Moines Area Community College                 1
Beth Medrash Govoha of America                    1
Columbia Southern University                      1
Presidio Graduate School                          1
Name: count, Length: 318, dtype: int64


In [85]:
uni_avg = total_by_uni / total_donations_uni
print(uni_avg.median())
print(total_by_uni.median())

500000.0
6486791.5


In [95]:
print(df_sorted_uni['Foreign Gift Amount'].mean().median())
print('\n')
print(df_sorted_uni['Foreign Gift Amount'].median().median())

500000.0


364119.5


In [103]:
country_to_university = df.groupby(['Country of Giftor', 'Institution Name'])
country_to_university['Foreign Gift Amount'].sum().sort_values(ascending = False)

Country of Giftor     Institution Name                     
QATAR                 Cornell University                       1018473315
BERMUDA               Carnegie Mellon University                750000000
QATAR                 Texas A&M University                      504099350
                      Carnegie Mellon University                425410702
                      Georgetown University                     327128310
                                                                  ...    
CAYMAN ISLANDS (THE)  Johns Hopkins University                         10
TRINIDAD              Westminster Theological Seminary                 10
GERMANY               Westminster Theological Seminary                 10
ERITREA               University of California, Los Angeles             3
MEXICO                University of California, Merced                  1
Name: Foreign Gift Amount, Length: 2685, dtype: int64

In [81]:
df_sorted_name = df.groupby('Giftor Name')
total_donations_name = df_sorted_name['Giftor Name'].value_counts()
total_donations_name.sort_values(ascending = False).head(6)

Giftor Name
Saudi Arabian Cultural Mission    480
Human Frontier Science Program    287
Anonymous                         262
Qatar Foundation                  261
Novartis Pharmaceuticals Corp     251
Royal Embassy of Saudi Arabia     210
Name: count, dtype: int64

1. Qatar donates the most by total value, with $2,706,240,869 donated.
2. England gives the most gifts by count, with 3655 total gifts.
3. Bermuda gives an average of $7,688,837, making it the country with the largest average donation.  
4. Carnegie Mellon University receives the most money in total.
5. The institution that receives the most gifts by count is UCLA.
6. The median of the average of donations is $500,000, and the median donation is $364,119.50. The mean donation is larger than the median due to some extremely large donations, making it's value larger and causing it to be right skewed.
7. The largest flow of donations from one country-university pairing from Qatar to Cornell at $1,018,473,315.
8. The top 5 giftors to academic institutions in the U.S. are the Saudi Arabian Cultural Mission, Human Frontier Science Program, Qatar Foundation, Novartis Pharmaceuticals Corp, and the Royal Embassy of Saudi Arabia.